site/_posts/2020-11-15-how-to-securely-implement-oauth-rails.md

```diff
@@ -0,0 +1,382 @@
+
+## Prerequisites
+- Rails 6
+- Docker (optional but preferred for installing FusionAuth)
+
+## Setting up FusionAuth
+In this tutorial we will use Docker for quick setup:
+```bash
+curl -o docker-compose.yml https://raw.githubusercontent.com/FusionAuth/fusionauth-containers/master/docker/fusionauth/docker-compose.yml
+curl -o .env https://raw.githubusercontent.com/FusionAuth/fusionauth-containers/master/docker/fusionauth/.env
+docker-compose up
+```
+
+If you prefer to install FusionAuth locally, see the [5-Minute Setup Guide](https://fusionauth.io/docs/v1/tech/5-minute-setup-guide/) to get up and running.  
+
+Once you have FusionAuth installed, it should be running at [localhost:9011](http://localhost:9011). Navigate there and finish the setup steps. 
+When those are completed, login as the administrator that you just created. Now we are ready to rock!
+
+A [FusionAuth Application](https://fusionauth.io/docs/v1/tech/core-concepts/applications/) is simply something a user can log in to.
+As such, we want to create a new application in FusionAuth for our Rails app.
+
+### Create an application
+Select "Applications" from the left nav bar. Then click the green plus sign in the upper-right hand corner to add a new application.
+We will want to configure a few pieces of important information:
+* **Name** (Required): A display name for our application. Enter `fusionauth-rails-app`.
+* **Authorized redirect URL's**: Found under the "OAuth" tab. This URL tells FusionAuth where to redirect to after a user successfully authenticates. 
+* **Logout URL**: Also found under the "OAuth" tab. This URL tells FusionAuth where to redirect to when a user logs out.
+
+{% include _image.liquid src="/assets/img/blogs/fusionauth-example-rails/add-application.png" alt="Creating a new Application in FusionAuth." class="img-fluid" figure=false %}
+
+### Create a user
+For this example, we will manually create a user. Click on "Users" in the left-nav bar and then click the green plus sign in the upper-right hand corner to add a new user.
+
+{% include _image.liquid src="/assets/img/blogs/fusionauth-example-rails/create-user.png" alt="Creating a new User in FusionAuth." class="img-fluid" figure=false %}
+
+Configure the following attributes and then save the new user:
+* **Email**
+* **Password and Confirm**: Toggle `Send email to setup password` to off and manually enter a password for the user.
+
+### Register the user
+Looking good! Now we have our new application and a user. The last thing we want to do is register the user with our application.
+Navigate to "Users -> Manage" for our newly created user.
+
+Click on the "Add Registration" button, select the Rails application we created, and then click on the "Save" button.
+
+{% include _image.liquid src="/assets/img/blogs/fusionauth-example-rails/user-registered.png" alt="The Manage User page in FusionAuth." class="img-fluid" figure=false %}
```

Comment:
Nit about the images. I usually leave a few pixels of space between the red box and the input or form control or data I'm highlighting.  See https://fusionauth.io/assets/img/blogs/custom-admin-registration-form/adding-user-manager-role-to-user.png for an example.

Do you have the original images? If not, don't worry about it. If so, can you add a bit of spacing?

Reply:
Re-did the images to these specifications :)